File: src/modules/film/film-model.js
const mongoose = require("mongoose");

const filmSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    episode_id: {
      type: Number,
      default: 1.0,
    },
    opening_crawl: {
      type: String,
      trim: true,
    },
    director: {
      type: String,
      trim: true,
      default: "Agustin Chavero",
    },
    producer: {
      type: String,
      trim: true,
      default: "Agustin Chavero",
    },
    release_date: {
      type: String,
      trim: true,
    },
    characters: {
      type: [String],
    },
    planets: {
      type: [String],
    },
    starships: {
      type: [String],
    },
    vehicles: {
      type: [String],
    },
    species: {
      type: [String],
    },
    is_deleted: {
      type: Boolean,
      default: false,
    },
  },
  {
    timestamps: true,
  }
);

const Film = mongoose.model("Film", filmSchema);

module.exports = Film;

File: csharp-asp/modules/film/film-model.cs
using MongoDB.Bson;

namespace csharp_asp.modules.film
{
    public class Film: IFilm
    {
        public string _id { get; set; }
        public string Title { get; set; } 
        public int? EpisodeId { get; set; }
        public string? OpeningCrawl { get; set; }
        public string? Director { get; set; }
        public string? Producer { get; set; } 
        public string? ReleaseDate { get; set; }
        public List<string>? Characters { get; set; }
        public List<string>? Planets { get; set; }
        public List<string>? Starships { get; set; }
        public List<string>? Vehicles { get; set; }
        public List<string>? Species { get; set; }
        public Boolean IsDeleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public Film()
        {
            _id = ObjectId.GenerateNewId().ToString();
            Title = "NewFilm";
            EpisodeId = 1;
            OpeningCrawl = "SuperOpening";
            Director = "Agustin Chavero";
            Producer = "Agustin Chavero";
            IsDeleted = false;
            ReleaseDate = DateTime.Now.ToString();
        }
    }
}
